clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -535,4 +536,46 @@ public static void setFieldValue(Object obj, String fieldName, Object value) thr
         field.setAccessible(true);
         field.set(obj, value);
     }
+
+    /**
+     * Returns true if both iterators have same elements in the same order.
+     *
+     * @param iterator1 first iterator.
+     * @param iterator2 second iterator.
+     * @param <T>       type of element in the iterators.
+     */
+    public static <T> boolean sameElementsWithOrder(Iterator<T> iterator1,
```

Comment:
Here is a slightly simpler version:
```
 while (iterator1.hasNext() && iterator2.hasNext()) {
     if (!Objects.equals(iterator1.next(), iterator2.next())) {
            return false;
    }
}

return !iterator1.hasNext() && !iterator2.hasNext();
```

Reply:
Existing code looks easy to read/comprehend and no multiple calls to hasNext().
How about the below code after removing inline variables in the existing code?

```
while (iterator1.hasNext()) {
    if (!iterator2.hasNext()) {
        return false;
    }

    if (!Objects.equals(iterator1.next(), iterator2.next())) {
        return false;
    }
}

return !iterator2.hasNext();
```

Reply:
Sounds good